gnosis-imap/gnosis-imap-spec.ini

```diff
@@ -72,7 +72,137 @@ localMem: M
+    ;andBool #rangeBytes( DATA_LEN,  DATA)
+    andBool #rangeUInt(   8,        OPERATION)
+    ; enum Enum.Operation, 3 possible values encoded to 0-2.
+    andBool OPERATION <=Int 2
+    andBool #rangeUInt( 256,        SAFE_TX_GAS)
+    andBool #rangeUInt( 256,        DATA_GAS)
+    andBool #rangeUInt( 256,        GAS_PRICE)
+    andBool #rangeAddress(          GAS_TOKEN)
+    andBool #rangeAddress(          REFUND_RECEIVER)
+    andBool #rangeUInt( 256,        NONCE)
+
+    andBool #rangeBytes( 32,        DOMAIN_SEPARATOR)
+
+    ; TODO more precise value. 1000 is too little.
+    andBool #range(0 <= #sizeWordStack(WS) <= 900)
+
+    andBool DATA_LEN <Int 2 ^Int 16
+
+    andBool SAFE_TX_HASH ==Int keccak( #encodeArgs(
+                                 #bytes32(#parseHexWord({SAFE_TX_TYPEHASH})),
+                                 #address(TO),
+                                 #uint256(VALUE),
+                                 #bytes32(keccak(#buf(DATA_LEN, DATA))),
+                                 #uint8(OPERATION),
+                                 #uint256(SAFE_TX_GAS),
+                                 #uint256(DATA_GAS),
+                                 #uint256(GAS_PRICE),
+                                 #address(GAS_TOKEN),
+                                 #address(REFUND_RECEIVER),
+                                 #uint256(NONCE) ))
+    andBool INIT_MEM ==K   storeRange( storeRange( storeRange( storeRange( _,
+                              64,                   32,       #buf(32, INIT_USED_MEM_PTR) ),
+                              DATA_LEN_LOC,         32,       #buf(32, DATA_LEN) ),
+                              DATA_BUF_LOC,         DATA_LEN, #buf(DATA_LEN, DATA) ),
+                              INIT_USED_MEM_ACTUAL, 32,       #padToWidth(32, #asByteStack(0)) )
+    andBool INIT_USED_MEM_PTR  ==Int #ceil32(DATA_LEN) +Int DATA_LEN_LOC +Int 32
+
+    ; local memory ranges
+    ; on encodeTransactionData-public:
+    ; andBool DATA_LEN_LOC ==Int 128
+    ; DATA_LEN_LOC is the base memory location for all other locations.
+    ; We give a wider range enough to cover all execution paths.
+    andBool #range(128 <= DATA_LEN_LOC <= 2 ^Int 16)
+    andBool DATA_LEN_LOC modInt 32 ==Int 0
+    andBool DATA_BUF_LOC  ==Int DATA_LEN_LOC +Int 32
+    andBool STATIC_MU     ==Int DATA_LEN_LOC +Int 32
+    ; required to resolve select...store expressions.
+    andBool INIT_USED_MEM_ACTUAL ==Int DATA_LEN_LOC +Int 32 +Int DATA_LEN
+    andBool #range(STATIC_MU <= INIT_USED_MEM_ACTUAL <= INIT_USED_MEM_PTR)
+
+    andBool START_MEM_USED ==Int #memoryUsageUpdate(STATIC_MU /Int 32, INIT_USED_MEM_ACTUAL,       32)
+    andBool END_MEM_USED   ==Int #memoryUsageUpdate(STATIC_MU /Int 32, INIT_USED_MEM_PTR +Int 450, 32)
+
+    ; storage
+    andBool select(STORAGE, #hashedLocation({COMPILER}, {DOMAIN_SEPARATOR}, .IntList)) ==Int DOMAIN_SEPARATOR
+
+[encodeTransactionData-internal-proof]
++attribute: [matching(_:_WS,_++_WS,storeRange)]
```

Comment:
don't you need `#buf` as well?

Reply:
I don't need any of them actually. With the changes to KEVM these [matching] are optional. But with unchanged KEVM they would be needed. Yet there's no way to avoid KEVM changes for proof spec.

Reply:
could you explain why "there's no way to avoid KEVM changes for proof spec"?

Reply:
See this post: https://runtimeverification.slack.com/archives/CA1CQKLMQ/p1546091259027800?thread_ts=1546026489.019300&cid=CA1CQKLMQ